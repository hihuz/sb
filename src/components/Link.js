import React from "react";

function Link({ url, name, viewbox, path }) {
  return (
    <div className="link-container">
      <a href={url} title={name}>
        <svg viewBox={viewbox} role="presentation" focusable="false">
          <path d={path} />
        </svg>
      </a>
    </div>
  );
}

export default Link;
