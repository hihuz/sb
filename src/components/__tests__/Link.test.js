import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Link from "../Link";

describe("Link", () => {
  test("Snapshot", () => {
    // eslint-disable-next-line
    const component = shallow(<Link url="http://some.url" />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });
});
