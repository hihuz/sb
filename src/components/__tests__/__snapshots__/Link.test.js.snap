// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Snapshot 1`] = `
<div
  className="link-container"
>
  <a
    href="http://some.url"
  >
    <svg
      focusable="false"
      role="presentation"
    >
      <path />
    </svg>
  </a>
</div>
`;
